src/renderer/features/operation/sign/ui/VaultSigning/VaultSigning.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from 'react';
+import { UnsignedTransaction } from '@substrate/txwrapper-polkadot';
+
+import { SigningProps } from '@renderer/features/operation';
+import { useCountdown } from '@renderer/shared/lib/hooks';
+import ScanMultiframeQr from '@renderer/components/common/Scanning/ScanMultiframeQr';
+import ScanSingleframeQr from '@renderer/components/common/Scanning/ScanSingleframeQr';
+import { ValidationErrors } from '@renderer/shared/lib/utils';
+import { useTransaction } from '@renderer/entities/transaction';
+import { HexString } from '@renderer/domain/shared-kernel';
+import QrReaderWrapper from '@renderer/components/common/QrCode/QrReader/QrReaderWrapper';
+
+export const VaultSigning = ({
+  chainId,
+  api,
+  addressPrefix,
+  validateBalance,
+  onGoBack,
+  accounts,
+  signatory,
+  transactions,
+  onResult,
+}: SigningProps) => {
+  const { verifySignature } = useTransaction();
+
+  const [countdown, resetCountdown] = useCountdown(api);
+  const [unsignedTxs, setUnsignedTxs] = useState<UnsignedTransaction[]>([]);
+  const [txPayloads, setTxPayloads] = useState<Uint8Array[]>([]);
+
+  const [validationError, setValidationError] = useState<ValidationErrors>();
+
+  const isScanStep = !unsignedTxs.length && !txPayloads.length;
+  const isMultiframe = transactions.length > 1;
+
+  useEffect(() => {
+    if (countdown === 0) {
+      scanAgain();
+    }
+  }, [countdown]);
+
+  const handleSignature = async (data: string | string[]) => {
+    const isMultishard = Array.isArray(data);
+    const signatures = isMultishard ? (data as HexString[]) : [data as HexString];
+    const accountIds = isMultiframe ? accounts.map((t) => t.accountId) : [(signatory || accounts[0])?.accountId];
+
+    const isVerified = signatures.every((signature, index) => {
+      // TODO: Research complex verification
+      // TODO: research multishard signature verification
+      if (isMultishard) return true;
```

Comment:
How did we verify in the previously?

Reply:
Just verify every signature doesn't work in some cases, and we decided to temporarily disable it